src/fields/lib/abstract-field.js

```diff
@@ -236,4 +247,105 @@ module.exports = class AbstractField {
       value === 'undefined' ? undefined :
         value;
   }
+
+  /**
+   * Checks for read permission on the field. Override or pass in constructor options
+   * this method for field level authentication. Defaults to true otherwise
+   *
+   * @method canRead
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   */
+  canRead(bundle) {
+    if (this._canRead) {
+      return Promise.resolve(this._canRead.call(null, bundle))
+        .then(result => {
+          if (!result) {
+            return Promise.reject(new Restypie.TemplateErrors.FieldNotReadable({
+              key: this.key
+            }));
+          }
+        });
+    }
+    return Promise.resolve(true);
+  }
+
+  /**
+   * Checks for create permission on the field. Override or pass in constructor options
+   * this method for field level authentication. Defaults to true otherwise
+   *
+   * @method canWriteOnCreate
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   */
+  canWriteOnCreate(bundle) {
+    if (this._canWriteOnCreate) {
+      return Promise.resolve(this._canWriteOnCreate.call(null, bundle))
+        .then(result => {
+          if (!result) {
+            return Promise.reject(new Restypie.TemplateErrors.FieldNotWritable({
+              key: this.key
+            }));
+          }
+        });
+    }
+    return Promise.resolve(true);
+  }
+
+  /**
+   * Checks for update permission on the field. Override or pass in constructor options
+   * this method for field level authentication. Defaults to true otherwise
+   *
+   * @method canWriteOnUpdate
+   * @param bundle
+   * @returns {Promise.<boolean>}
+   */
+  canWriteOnUpdate(bundle) {
+    if (this._canWriteOnUpdate) {
+      return Promise.resolve(this._canWriteOnUpdate.call(null, bundle))
+        .then(result => {
+          if (!result) {
+            return Promise.reject(new Restypie.TemplateErrors.FieldNotUpdatable({
```

Comment:
Rejecting here forces the override the throw the error as well. A way to fix this is add another private method of each of these that throws the error.